qiime/core/type/primitive.py

```diff
@@ -349,6 +349,24 @@ def encode(self, value):
 Float = _Float('Float')
 
 
+class _Bool(_Primitive):
+    _valid_predicates = {Arguments,}
+
+    def _is_element_(self, value):
+        return isinstance(value, bool)
+
+    def decode(self, string):
+        if string not in ('false', 'true'):
+            raise TypeError("%s is neither 'true' or 'false'")
+
+        return string == 'true'
+
+    def encode(self, value):
+        return str(value)
```

Comment:
Right, but Int.encode(1.234) -> is float

On Wed, Aug 17, 2016 at 4:24 PM, Jai Ram Rideout <notifications@github.com>
wrote:

> In qiime/core/type/primitive.py
> <https://github.com/qiime2/qiime2/pull/116#discussion_r75225855>:
>
> > @@ -349,6 +349,24 @@ def encode(self, value):
> >  Float = _Float('Float')
> >
> >
> > +class _Bool(_Primitive):
> > +    _valid_predicates = {Arguments,}
> > +
> > +    def _is_element_(self, value):
> > +        return isinstance(value, bool)
> > +
> > +    def decode(self, string):
> > +        if string not in ('false', 'true'):
> > +            raise TypeError("%s is neither 'true' or 'false'")
> > +
> > +        return string == 'true'
> > +
> > +    def encode(self, value):
> > +        return str(value)
>
> str(42) returns '42', which is the correct JSON representation of an
> integer. str(True) returns 'True', which is not a valid JSON boolean
> type. Does that make sense? encode/decode really need to be renamed
> to_json/from_json.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/qiime2/qiime2/pull/116/files/bdb82d4db3ea397a8c42fc00f0e4be1ace7ff98e#r75225855>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAc8ssk1Y0h5NSMMLCX3UbPa-u1-SaWsks5qg5g_gaJpZM4Jlp9U>
> .
>